### PR TITLE
fix(orc8r): Log entries created from user input

### PR DESCRIPTION
**Pull request by spikey979**
fix(orc8r): Log entries created from user input

## Summary

Added function for string sanitizing before it is written to a log entry.
function works as follows:
it escape special characters in HTML text,
it remove line breaks,
it remove extra whitespace

## Test Plan

I ran tests /magma/orc8r/cloud/docker/  ./build.py -c

This is for alert:
https://github.com/magma/magma/security/code-scanning/40?query=ref%3Arefs%2Fheads%2Fmaster

### CODE REVIEW

**Comment by spikey979** [comment]
Can anyone tell me why i get the error: Changes not staged for commit?

File: orc8r/cloud/go/obsidian/server/metrics.go [hunk 1]
```diff
@@ -50,12 +50,21 @@ func CollectStats(next echo.HandlerFunc) echo.HandlerFunc {
...
 	}
 }
+
+func sanitizeString(strString string) string {
+	strSanitizedString := html.EscapeString(strString) //escape special charatchters in HTML text
+	strSanitizedString = strings.Replace(strSanitizedString, "\r", "", -1) //remove line breaks
+	strSanitizedString = strings.Replace(strSanitizedString, "\n", "", -1)
+	strSanitizedString = strings.Join(strings.Fields(strSanitizedString), " ") //remove extra whitespace
+	return strSanitizedString
+}
```
Comment (uri200) [inline orc8r/cloud/go/obsidian/server/metrics.go:68]: when did you see an url containing `\r` or `\n`?  If that is the case maybe there are another issue here

Comment (spikey979) [reply orc8r/cloud/go/obsidian/server/metrics.go:68]: I threw out  strings.Replace for \r and \n...

**Comment by lucasgonze** [comment]
@spikey979 After talking this over with Eli, I propose these followups:

1. Create a library routine SafeLog that is shared across all code that logs input from untrusted sources. Input would be a single string. Output would be limited to 1024 chars and would be URI-encoded to prevent embedded commands or XSS attacks from being evaluated. It would be used instead of glog.V(2).Infof and would accept the same parameters.
2. Create a single PR fixing all nine alerts. Close the three open PRs [10934](https://github.com/magma/magma/pull/10934), [10939](https://github.com/magma/magma/pull/10939), and [11269](https://github.com/magma/magma/pull/11269).

Thoughts?

**Pull request closed without merging**